cl/lib/zoho.py

```diff
@@ -48,8 +123,82 @@ def initialize():
             resource_path=settings.ZOHO_RESOURCE_PATH,
         )
 
-    def get_record_by_cl_id_or_email(
+    @staticmethod
+    def handle_api_response(response):
+        """
+        Handle a Zoho API response, raising exceptions on errors.
+
+        :param response: The response object from a Zoho API call.
+        :return: The data from the response if successful.
+        :raises Exception: if the response is None, empty, or contains an APIException.
+        """
+        if response is None:
+            raise Exception("Received no response from the API.")
+
+        status_code = response.get_status_code()
+        if status_code in [204, 304]:
+            msg = "No Content" if status_code == 204 else "Not Modified"
+            logger.info(f"Zoho API returned no records ({msg}).")
+            return []
+
+        response_object = response.get_object()
+        if response_object is None:
+            raise Exception("Zoho API returned an empty response object.")
+
+        if isinstance(response_object, ResponseWrapper | ActionWrapper):
+            return response_object.get_data()
+
+        if isinstance(response_object, APIException):
+            status = response_object.get_status().get_value()
+            code = response_object.get_code().get_value()
+            message = response_object.get_message().get_value()
+            details = response_object.get_details()
+
+            detail_str = ", ".join(f"{k}: {v}" for k, v in details.items())
+            raise Exception(
+                f"Zoho API Exception [{code}] {status}: {message} | Details: {detail_str}"
+            )
+
+        raise Exception("Unexpected response type received from the Zoho API.")
+
+    def _build_record(
+        self, fields: dict[str | Field, Any], record_id: int | None = None
+    ) -> Record:
+        """
+        Build a Zoho Record instance from a dictionary of fields.
+        Optionally set a record ID for updates.
+        """
+        record = Record()
+        if record_id:
+            record.set_id(record_id)
+
+        for key, value in fields.items():
+            if isinstance(key, Field):
+                record.add_field_value(key, value)
+            else:
+                record.add_key_value(key, value)
+        return record
+
+    def _build_body_wrapper(
```

Comment:
Same suggestion as above: can be an `@staticmethod` so we can remove `self`. Also in `HasModuleName` typing class.